keras_nlp/layers/modeling/cached_multi_head_attention.py

```diff
@@ -101,17 +103,18 @@ def call(
         # cache at the specified index. `cache = None` handles the training
         # case, where we don't use the cache at all.
         if cache is not None:
-            key_cache, value_cache = tf.unstack(cache, axis=1)
+            key_cache = cache[:, 0, ...]
```

Comment:
(optional): ops.split might be a bit nicer here

Reply:
Was not able to get this working. `split(cache, 2, axis=1)` is not specific enough shape wise. Is that the right invocation you were thinking of?

Reply:
Ahh yes, that probably won't work then. `split(cache, [1], axis=1)` would probably give you a defined shape for `key_cache` but not `value_cache`. So this seems good as-is